### PR TITLE
fix(authentication): fixed login and call back redirect

### PR DESCRIPTION
**Pull request by phoenixdahdev**
# Description

**Closes #issue_number_here**

# Changes Proposed

## What were you told to do?

Implement credential sign-in using NextAuth.
decleared global user
Fixed routing on login 
Also added error boundaries 

## What did you do?

- Implemented credential sign-in functionality using NextAuth.
- Modified authentication handler to support email/password sign-in.
- Updated related components and pages to integrate with the new authentication flow.
- Added configuration for NextAuth in the project.

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR are clear and explain the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit message style matches the requested structure.
- [x] My code additions will not fail code linting checks or unit tests.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos

---
Screenshot_2024-07-31_11-05-28

### CODE REVIEW

File: .env.sample
Comment (prudentbird) [inline .env.sample:1]: Undo this change

File: package.json
Comment (PrudentBird) [inline package.json:1]: Undo all changes to package json

File: pnpm-lock.yaml
Comment (prudentbird) [inline pnpm-lock.yaml:1]: Undo all changes to pnpm lock

File: src/actions/register.ts
Comment (prudentbird) [inline src/actions/register.ts:1]: Send register file to utils from actions

File: src/config/auth.config.ts
Comment (prudentbird) [inline src/config/auth.config.ts:1]: Extract type declarations to src/types

File: src/lib/auth.react.ts
Comment (prudentbird) [inline src/lib/auth.react.ts:1]: Fix the type issues instead

**Review by prudentbird — CHANGES_REQUESTED** [review]

File: package.json
Comment (prudentbird) [inline package.json:1]: Use lucide-react or import the svg

No other icon library in use, dependencies are too much

File: src/app/error.tsx
Comment (PrudentBird) [inline src/app/error.tsx:1]: Just one change remaining

**Review by prudentbird — DISMISSED** [review]

**Review by prudentbird — APPROVED** [review]

**Review by billmal071 — APPROVED** [review]
LGTM
